src/Microsoft.Security.Utilities/CustomAlphabetEncoder.cs

```diff
@@ -115,5 +115,41 @@ public byte[] Decode(string encodedValue)
 
             return BitConverter.GetBytes(decodedValue);
         }
+        
+        /// <summary>
+        /// Sets a custom alphabet to be used for all encoding and decoding operations.
+        /// </summary>
+        /// <param name="customAlphabet">A string representing the custom alphabet. It must consist of non-whitespace ASCII characters, including letters, numbers, and/or punctuation.</param>
+        /// <exception cref="ArgumentException">Throws an exception if the alphabet contains duplicates or contains forbidden characters.</exception>
+        /// <remarks>
+        /// This method allows dynamic changes to the alphabet used for encoding and decoding data. Before setting a new alphabet, the method performs checks for duplicates and the presence of forbidden characters.
+        /// </remarks>
+        public void SetCustomAlphabet(string customAlphabet)
```

Comment:
>SetCustomAlphabet [](http://example.com/codeflow?start=20&length=17)

Sorry for not noticing this PR earlier! This looks like a good idea. It'd be best to share the argument validated logic between this method and the constructor via a shared helper.